lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -361,16 +361,8 @@ public function getListTableIndexesSQL($table, $currentDatabase = null)
      */
     private function getTableWhereClause($table, $classAlias = 'c', $namespaceAlias = 'n')
     {
-        $whereClause = $namespaceAlias.".nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast') AND ";
-        if (strpos($table, ".") !== false) {
-            list($schema, $table) = explode(".", $table);
-            $schema = "'" . $schema . "'";
-        } else {
-            $schema = "ANY(string_to_array((select replace(replace(setting,'\"\$user\"',user),' ','') from pg_catalog.pg_settings where name = 'search_path'),','))";
-        }
-
-        $table = new Identifier($table);
-        $whereClause .= "$classAlias.relname = '" . $table->getName() . "' AND $namespaceAlias.nspname = $schema";
+        $tableNameEscaped = str_replace("'", "''", $table);
+        $whereClause = $classAlias.".oid='".$tableNameEscaped."'::regclass::oid";
```

Comment:
@deeky666 Incoming quoted names aren't handled well in Identifier... the full and proper fix needs to occur in 3.0 I think.